ui/admin/app/routes/scopes/scope/auth-methods/index.js

```diff
@@ -4,10 +4,103 @@
  */
 
 import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
 
 export default class ScopesScopeAuthMethodsIndexRoute extends Route {
+  // =services
+
+  @service store;
+  @service can;
+  @service router;
+
+  // =attributes
+
+  queryParams = {
+    search: {
+      refreshModel: true,
+      replace: true,
+    },
+    types: {
+      refreshModel: true,
+      replace: true,
+    },
+    primary: {
+      refreshModel: true,
+      replace: true,
+    },
+    page: {
+      refreshModel: true,
+    },
+    pageSize: {
+      refreshModel: true,
+    },
+  };
+
   // =methods
 
+  /**
+   * Loads queried auth-methods and the number of auth-methods under current scope.
+   * @returns {Promise<{totalItems: number, authMethods: [AuthMethodModel], authMethodsExist: boolean }> }
+   */
+  async model({ search, types, primary, page, pageSize }) {
+    const scope = this.modelFor('scopes.scope');
+    const { id: scope_id } = scope;
+    const filters = {
+      scope_id: [{ equals: scope_id }],
+      type: [],
+      is_primary: [],
+    };
+
+    types.forEach((type) => {
+      filters.type.push({ equals: type });
+    });
+    if (primary.length == 1) {
+      primary.forEach((val) => {
+        filters.is_primary.push({ equals: val });
+      });
+    }
+
+    let authMethods;
+    let totalItems = 0;
+    if (this.can.can('list model', scope, { collection: 'auth-methods' })) {
+      authMethods = await this.store.query('auth-method', {
+        query: { search, filters },
+        page,
+        pageSize,
+      });
+      totalItems = authMethods.meta?.totalItems;
+    }
+    const authMethodsExist = await this.getAuthMethodsExist(
+      scope_id,
+      totalItems,
+    );
+
+    return { authMethods, authMethodsExist, totalItems };
+  }
+
+  /**
+   * Sets authMethodsExist to true if there exists any auth methods.
+   * @param {string} scope_id
+   * @param {number} totalItems
+   * @returns {boolean}
+   */
+  async getAuthMethodsExist(scope_id, totalItems) {
+    if (totalItems > 0) {
+      return true;
+    }
+    const options = { pushToStore: false };
+    const authMethod = await this.store.query(
+      'auth-method',
+      {
+        query: { filters: { scope_id: [{ equals: scope_id }] } },
+        page: 1,
+        pageSize: 1,
+      },
```

Comment:
I think we should discuss the usage of these functions. I'm not sure if ya'll are aware but `page` and `pageSize` aren't being sent over to the API so there's no difference in performance whether we request for all or just for one item. I think what we really need is a way to query indexed DB (and being able to add to the store) without having to make an extra request to the API.

Thoughts?

Reply:
Yes, that is a good point, I'll also give this some more thought. We can discuss this further during the poddle.

Reply:
For now ill this as is, and we can address this & refactor in a different PR.